src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSinkUnitTest.java

```diff
@@ -52,4 +62,18 @@ public void readsSinkTest(String inputBam, String outputFile) throws IOException
         Assert.assertEquals(rddParallelReads.count(), rddParallelReads2.count());
     }
 
-}
\ No newline at end of file
+    @Test(dataProvider = "loadReadsADAM", groups = "spark")
+    public void readsSinkADAMTest(String inputBam, String outputFile) throws IOException {
+        new File(outputFile).deleteOnExit();
+        JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+
+        ReadsSparkSource readSource = new ReadsSparkSource(ctx);
+        JavaRDD<GATKRead> rddParallelReads = readSource.getParallelReads(inputBam);
+        SAMFileHeader header = ReadsSparkSource.getHeader(ctx, inputBam);
+
+        ReadsSparkSink.writeReads(ctx, outputFile, rddParallelReads, header, ReadsWriteFormat.ADAM);
```

Comment:
Please file two tickets:
1) Check in parallel BAM (several shards) written from Spark (assign it to me)
2) Check in a ADAM Parquet file (assign it to yourself)

We need to improve these tests. If the test fails, its unclear if it's a reader or writer problem.